.github/workflows/ci.yml

```diff
@@ -504,6 +501,13 @@ jobs:
         uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
         with:
           submodules: true
+      - id: "gcp-auth"
+        name: "Authenticating to Google Cloud"
+        uses: "google-github-actions/auth@v1"
+        with:
+          token_format: "access_token"
+          credentials_json: "${{ secrets.IREE_OSS_GITHUB_RUNNER_BASIC_TRUST_SERVICE_ACCOUNT_KEY }}"
+          create_credentials_file: false
       - name: "Downloading build dir archive"
         run: gcloud storage cp "${BUILD_DIR_GCS_ARTIFACT}" "${BUILD_DIR_ARCHIVE}"
```

Comment:
@pzread could you help here?

`Error: google-github-actions/auth failed with: retry function failed after 4 attempts: the GitHub Action workflow must specify exactly one of "workload_identity_provider" or "credentials_json"! If you are specifying input values via GitHub secrets, ensure the secret is being injected into the environment. By default, secrets are not passed to workflows triggered from forks, including Dependabot.`
https://github.com/openxla/iree/actions/runs/7644381985/job/20828896913?pr=16200#step:3:18

I'm trying to switch a few of these jobs to use GitHub-hosted runners, but I'm not sure how to authenticate for `gcloud storage cp` to pull the build archive from previous jobs. Is that even possible?

If it isn't, these sorts of changes may require switching to using GitHub for storage (of just packages, and not entire build archives), which I'd like to do anyways.
https://github.com/openxla/iree/blob/9edf1fc1520f0f050a34e976562887baff792698/.github/workflows/pkgci_build_packages.yml#L73-L79 https://github.com/openxla/iree/blob/9edf1fc1520f0f050a34e976562887baff792698/.github/workflows/pkgci_regression_test_cpu.yml#L42-L45

Reply:
Here are the logs when I don't try to authenticate: https://github.com/openxla/iree/actions/runs/7643847623/job/20827392052?pr=16200#step:3:12

```
Run gcloud storage cp "${BUILD_DIR_GCS_ARTIFACT}" "${BUILD_DIR_ARCHIVE}"
  

ERROR: (gcloud.storage.cp) You do not currently have an active account selected.
Please run:

  $ gcloud auth login

to obtain new credentials.
```

Reply:
Yeah, ... this is the problem of gcloud cli. It always requires a credential even the bucket is public. I think the workaround is switch to `wget https://storage.googleapis.com/...`

Reply:
`wget` works to download public files, but the standard runners don't have enough disk space for the uncompressed archives (and Docker files), even with a bit more file pruning. I might abandon this change and jump straight to package-based workflows.

Reply:
I don't believe there is any hope for salvaging the current approach, so +1 on a reset.